### PR TITLE
Vite: Fix static source handling for addon-docs

### PR DESCRIPTION
**Pull request by joshwooding**
Issue: #19882 #19554

## What I did

- [x] Add csf-plugin

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/lib/builder-vite/src/plugins/csf-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { Plugin } from 'vite';
```
Comment (joshwooding) [inline code/lib/builder-vite/src/plugins/csf-plugin.ts:1]: Kept a plugin so vite-config stayed clean

File: code/lib/builder-vite/src/plugins/csf-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export async function csfPlugin(config: ExtendedOptions): Promise<Plugin> {
+  const { presets } = config;
+
+  const addons = await presets.apply<StorybookConfig['addons']>('addons', []);
+  const docsOptions =
+    // @ts-expect-error - not sure what type to use here
+    addons.find((a) => [a, a.name].includes('@storybook/addon-docs'))?.options ?? {};
+
+  return vite(docsOptions?.csfPluginOptions);
```
Comment (IanVS) [inline code/lib/builder-vite/src/plugins/csf-plugin.ts:14]: It feels to me like `@storybook/csf-plugin` itself should take care of this...

**Review by IanVS — COMMENTED** [review]
This really simplifies things nicely!  The only part I'm not sure about is getting the settings from addon-docs.  @shilman is this something that `@storybook/csf-plugin` itself should be doing, maybe?

File: code/lib/builder-vite/src/plugins/csf-plugin.ts (hunk 2 above)
Comment (shilman) [reply code/lib/builder-vite/src/plugins/csf-plugin.ts:14]: I don't think csf-plugin should know about the user's main.js and it makes sense for addon-docs to glue things together like this. Hopefully this all goes away in 7.x with the annotations server & simplifies things even further.

**Review by shilman — APPROVED** [review]
Looks great! 💯

**Pull request merged**